core/sql/cli/Context.cpp

```diff
@@ -3850,6 +3856,48 @@ void ContextCli::createMxcmpSession()
   mxcmpSessionInUse_ = TRUE;
 }
 
+// ----------------------------------------------------------------------------
+// Method:  updateMxcmpSession
+//
+// Updates security attributes in child arkcmp
+//
+// Returns: 0 = succeeded; -1 = failed
+// ----------------------------------------------------------------------------
+Int32 ContextCli::updateMxcmpSession()
+{
+  // If no child arkcmp, just return
+  if (getArkcmp()->getServer() == NULL)
+    return 0;
+
+  // Send changed user information to arkcmp process
+  CmpContext *cmpCntxt = CmpCommon::context();
+  ex_assert(cmpCntxt, "No compiler context exists");
+  NABoolean authOn = cmpCntxt->isAuthorizationEnabled();
+
+  // The message contains the following:
+  //   (auth state and user ID are delimited by commas)
+  //     authorization state (0 - off, 1 - on)
+  //     integer user ID
+  //     database user name
+  // See CmpStatement::process (CmpMessageDatabaseUser) for more details
+  Int32 userAsInt = (Int32) databaseUserID_;
+  char userMessage [MAX_AUTHID_AS_STRING_LEN + 1 + MAX_USERNAME_LEN + 1 + 2];
+  str_sprintf(userMessage, "%d,%d,%s", authOn, userAsInt, databaseUserName_);
+  char *pMessage = (char *)&userMessage;
+
+  // Send message to child arkcmp, if one exists
+  ExSqlComp::ReturnStatus cmpStatus;
+  ExSqlComp *exSqlComp = getArkcmp();
```

Comment:
I noticed that the indexIntoCompilerArray is sometimes used. But if there is only one arkcmp associated with the process, there will only be entry 0.  Several other places in the code just call getArkcmp() by itself without using indexIntoCompilerArray.